var/spack/repos/builtin/packages/tandem/package.py

```diff
@@ -41,6 +44,13 @@ class Tandem(CMakePackage):
     variant("libxsmm", default=False, description="Install libxsmm-generator")
 
     depends_on("mpi")
+
+    for var in ["openmpi", "mpich", "mvapich", "mvapich2", "mvapich2-gdr"]:
+        depends_on(f"{var} +cuda", when=f"+cuda ^[virtuals=mpi] {var}")
+
+    for var in ["mpich", "mvapich2-gdr"]:
+        depends_on(f"{var} +rocm", when=f"+rocm ^[virtuals=mpi] {var}")
```

Comment:
These directives will never be triggered, since `tandem` doesn't have neither `rocm` nor `cuda` as variants.

Reply:
Thanks, I forgot to derive tandem of CudaPackage and  ROCmPackage, now it is fixed.